hoomd/write/gsd.py

```diff
@@ -212,20 +234,56 @@ def write(state, filename, filter=All(), mode='wb', log=None):
         writer = _hoomd.GSDDumpWriter(state._cpp_sys_def, Periodic(1), filename,
                                       state._get_group(filter), mode, False)
 
-        if log is not None:
-            writer.log_writer = _GSDLogWriter(log)
+        if all((logger is not None, log is not None)):
```

Comment:
```suggestion
        if logger is not None and log is not None:
```
With two conditions this is clearer.